cmd/kyma/provision/gardener/az/cmd.go

```diff
@@ -0,0 +1,200 @@
+package az
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/kyma-project/cli/internal/kube"
+
+	retry "github.com/avast/retry-go"
+	hf "github.com/kyma-incubator/hydroform/provision"
+	"github.com/kyma-incubator/hydroform/provision/types"
+	"github.com/kyma-project/cli/internal/cli"
+	"github.com/kyma-project/cli/internal/files"
+	"github.com/spf13/cobra"
+)
+
+type command struct {
+	opts *Options
+	cli.Command
+}
+
+//NewCmd creates a new az command
+func NewCmd(o *Options) *cobra.Command {
+	c := command{
+		Command: cli.Command{Options: o.Options},
+		opts:    o,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "az",
+		Short: "Provisions a Kubernetes cluster using Gardener on Azure.",
+		Long: `Use this command to provision Kubernetes clusters with Gardener on Azure for Kyma installation. 
+To successfully provision a cluster on Azure, you must first create a service account to pass its details as one of the command parameters. 
+Create a service account with the ` + "`contributor`" + ` role. Use service account details to create a Secret and store it in Gardener.`,
+
+		RunE: func(_ *cobra.Command, _ []string) error { return c.Run() },
+	}
+
+	cmd.Flags().StringVarP(&o.Name, "name", "n", "", "Name of the cluster to provision. (required)")
+	cmd.Flags().StringVarP(&o.Project, "project", "p", "", "Name of the Gardener project where you provision the cluster. (required)")
+	cmd.Flags().StringVarP(&o.CredentialsFile, "credentials", "c", "", "Path to the kubeconfig file of the Gardener service account for a target provider. (required)")
+	cmd.Flags().StringVarP(&o.Secret, "secret", "s", "", "Name of the Gardener secret used to access the target provider. (required)")
+	cmd.Flags().StringVarP(&o.KubernetesVersion, "kube-version", "k", "1.16", "Kubernetes version of the cluster.")
+	cmd.Flags().StringVarP(&o.Region, "region", "r", "westeurope", "Region of the cluster.")
+	cmd.Flags().StringSliceVarP(&o.Zones, "zones", "z", []string{"1"}, "Zones specify availability zones that are used to evenly distribute the worker pool. eg. --zones=\"europe-west3-a,europe-west3-b\"")
+	cmd.Flags().StringVarP(&o.MachineType, "type", "t", "Standard_D8_v3", "Machine type used for the cluster.")
```

Comment:
I think the Standard_D4_v3 is enough. D8 is recommended for production profile only. Keep it consistent with the other provider settings taking a machine with 4CPU and 16RAM

Reply:
I adapted all to 4 CPU and 16GB Memory.